src/main/java/com/microsoft/aad/msal4j/DefaultHttpClient.java

```diff
@@ -99,6 +99,7 @@ private HttpResponse readResponseFromConnection(final HttpsURLConnection conn) t
             if (responseCode != HttpURLConnection.HTTP_OK) {
                 is = conn.getErrorStream();
                 if (is != null) {
+                    httpResponse.headers(conn.getHeaderFields());
```

Comment:
>ields()); [](start = 56, length = 9)

bug? Seems like there is a test missing?

Reply:
yes, bug - headers were not returned in case of error, missed test class during commit , will add